### PR TITLE
Wrote document of CRF1d argmax

### PR DESCRIPTION
**Pull request by unnonouno**
I wrote description of argmax function for linear-chain CRF, and made these methods public.
We need to consider about its interface.

### CODE REVIEW

File: tests/chainer_tests/functions_tests/loss_tests/test_crf1d.py [hunk 1]
```diff
@@ -96,6 +96,13 @@ def check_argmax(self, cost_data, xs_data):
         xs = [chainer.Variable(x) for x in xs_data]
         s, path = functions.loss.crf1d.argmax_crf1d(cost, xs)
 
+        isinstance(s, chainer.Variable)
+        isinstance(path, list)
```
Comment (gwtnb) [inline tests/chainer_tests/functions_tests/loss_tests/test_crf1d.py]: These `isinstance`s have no effect.

Comment (rezoo) [inline tests/chainer_tests/functions_tests/loss_tests/test_crf1d.py]: Ah sorry I missed it.

File: chainer/functions/loss/crf1d.py [hunk 2]
```diff
@@ -127,6 +127,32 @@ def crf1d(cost, xs, ys):
...
+    Args:
+        cost (Variable): A :math:`K \\times K` matrix which holds transition
+            cost between two labels, where :math:`K` is the number of labels.
+        xs (list of Variable): Input vector for each label.
+            ``len(xs)`` denotes the length of the sequence,
+            and each :class:`~chainer.Variable` holds a :math:`B \\times K`
+            matrix, where :math:`B` is mini-batch size, :math:`K` is the number
+            of labels.
+            Note that :math:`B` s in all the variables are not necessary
+            the same, i.e., it accepts the input sequences with difference
```
Comment (rezoo) [inline chainer/functions/loss/crf1d.py]: oops I made a typo. s/difference/different/

File: chainer/functions/loss/crf1d.py [hunk 3]
```diff
@@ -127,6 +127,32 @@ def crf1d(cost, xs, ys):
...
+            lengths.
+
+    Returns:
+        tuple: A tuple of :class:`~chainer.Variable` object ``s`` and a
+            :class:`list` ``ps``.
+            The shape of ``s`` is ``(B,)``, where ``B`` is the mini-batch size.
+            i-th element of ``s``, ``s[i]``, represents log-likelihood of i-th
+            data.
+            ``ps`` is a list of :class:`numpy.ndarray` or
+            :class:`cupy.ndarray`.
```
Comment (rezoo) [inline chainer/functions/loss/crf1d.py]: ``ps`` is a list of :class:`numpy.ndarray` or :class:`cupy.ndarray`, and denotes the state that maximizes the point probability.

**Comment by rezoo** [comment]
@unnonouno 20 days have passed since the last comment. I'll work instead of you if you don't have time .

**Comment by unnonouno** [comment]
Thank you for your reviewing very much, and sorry for late reply. I'll fix now!!

**Comment by rezoo** [comment]
Good. Can you deal with l.99 and l.100 in `test_crf1d.py`?

**Comment by gwtnb** [comment]
@unnonouno Could you fix this? https://github.com/pfnet/chainer/pull/1758#discussion_r83156457

**Comment by rezoo** [comment]
Thank you!! then I would like to merge it.

**Pull request merged**